### PR TITLE
🌱 Uplift CAPI to v1.3.0-rc.0 in main branch

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:
and also some minor missing additions for Ginkgo v2 migration

xref: https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.3.0-rc.0

### CODE REVIEW

**Review by kashifest — APPROVED** [review]

**Comment by furkatgofurov7** [comment]
/hold 
until https://github.com/metal3-io/metal3-dev-env/pull/1106 is merged

**Comment by kashifest** [comment]
Remember to squash the commits

**Comment by furkatgofurov7** [comment]
> Remember to squash the commits

Done.

> /hold
> until [metal3-io/metal3-dev-env#1106](https://github.com/metal3-io/metal3-dev-env/pull/1106) is merged

metal3-io/metal3-dev-env#1106](https://github.com/metal3-io/metal3-dev-env/pull/1106 is merged
/hold cancel
/test-ubuntu-integration-main
/test-ubuntu-e2e-main
/test-ubuntu-e2e-parallel-main
/test-upgrade-e2e-main

**Comment by furkatgofurov7** [comment]
/hold

e2e tests are not tested properly. 
https://jenkins.nordix.org/job/metal3_capm3_main_e2e_test_ubuntu/97/
https://jenkins.nordix.org/job/metal3_capm3_main_e2e_test_ubuntu_parallel/98/
https://jenkins.nordix.org/job/metal3_capm3_main_e2e_upgrade_test_ubuntu/133/

File: hack/gen_tilt_settings.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ function get_latest_release() {
     release="$(curl -H "Authorization: token ${GITHUB_TOKEN}" -sL "${1}")" || ( set -x && exit 1 )
   fi
   # This gets the latest release as vx.y.z , ignoring any version with a suffix starting with - , for example -rc0
-  release_tag="$(echo "$release" | jq -r "[.[].tag_name | select( startswith(\"${2:-""}\")) | select(contains(\"-\")==false)] | max ")"
+  release_tag="$(echo "$release" | jq -r "[.[].tag_name | select( startswith(\"${2:-""}\"))] | max ")"
```
Comment (lentzi90) [inline hack/gen_tilt_settings.sh:28]: nit: please fix the comment above so it follows the code

Comment (furkatgofurov7) [reply hack/gen_tilt_settings.sh:28]: ah, yes, it is unfortunate that I run e2e test which is 3h+ long...

Comment (furkatgofurov7) [reply hack/gen_tilt_settings.sh:28]: I will wait for it to pass, and re-run only integration test here after I fix it.

**Review by lentzi90 — COMMENTED** [review]
Thanks!
/lgtm

File: hack/gen_tilt_settings.sh (hunk 1 above)
Comment (furkatgofurov7) [reply hack/gen_tilt_settings.sh]: @lentzi90 fixed it, and re-run e2e again (the previous run failed due to an issue/flake unrelated to these changes)

Comment (lentzi90) [reply hack/gen_tilt_settings.sh]: Thanks!
(For some reason I cannot mark this thread as resolved...)

Comment (furkatgofurov7) [reply hack/gen_tilt_settings.sh]: Resolved now

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

**Pull request merged**